src/fklearn/types/types.py

```diff
@@ -18,6 +18,7 @@
 
 UncurriedLearnerFnType = Callable[..., LearnerReturnType]
 LearnerFnType = Callable[[pd.DataFrame], LearnerReturnType]
+LearnerMutableParametersFnType = Callable[..., LearnerFnType]
```

Comment:
Why do we need a new type here? It's not very clear to me

Reply:
Hey @prcastro 😄 I created this new type exclusively because the s-learner requires the creation of a new feature (T), that determines the treatment case (T=1) and the control case (T=0). Since we are creating this function upon fklearn, when we update the features from the classifier's partial function, mypy breaks, because the partial function only expects a dataframe as its input (it does not expect that you change the features parameter that were already provided). The solution I found was to create a more "flexible" type. But maybe there is a better way!

Here is where mypy breaks: https://github.com/nubank/fklearn/blob/a42339807e18a8cbdac5dc4fd8c568c01715ab5c/src/fklearn/causal/cate_learning/meta_learners.py#L229-L233

Reply:
The typing system in python don’t play well with partial function, that’s a fact. I’d do one of two things:

- Just declare the learner as `Callable` without specifying inputs and outputs
- Rename the type `PartialLearnerFnType` so it’s more clear

I prefer the first one.

Reply:
✅